server/matchmaker/search.py

```diff
@@ -173,8 +175,20 @@ def matches_with(self, other: 'Search'):
         """
         if not isinstance(other, Search):
             return False
-        elif self.quality_with(other) >= self.match_threshold and \
-                other.quality_with(self) >= other.match_threshold:
+
+        quality = self.quality_with(other)
+        return self._match_quality_acceptable(other, quality)
+
+    def _match_quality_acceptable(self, other: 'Search', quality: float) -> bool:
+        """
+        Determine if the given match quality is acceptable.
+
+        This gets it's own function so we can call it from the Matchmaker using
+        a cached `quality` value.
+        """
+        # NOTE: We are assuming for optimization purposes that quality is
+        # symmetric. If this ever changes, update here
+        if quality >= self.match_threshold and quality >= other.match_threshold:
```

Comment:
Why not simply return the condition of the `if` statement?
Or if it is more readable:
```python
is_acceptable = quality >= self.match_threshold and quality >= other.match_threshold
return is_acceptable
```

Reply:
Good point :P